core/src/main/java/io/methvin/watcher/visitor/DefaultFilesVisitor.java

```diff
@@ -0,0 +1,48 @@
+package io.methvin.watcher.visitor;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+public class DefaultFilesVisitor implements FilesVisitor {
+  public DefaultFilesVisitor(FailedCallback failedCallback) {
+    this.failedCallback = failedCallback;
+  }
+
+  private final FailedCallback failedCallback;
+
+  @Override
+  public void recursiveVisitFiles(Path file, Callback onDirectory, Callback onFile)
+      throws IOException {
+    SimpleFileVisitor<Path> visitor =
+        new SimpleFileVisitor<>() {
```

Comment:
we're building this for jdk8 so this doesn't work :(

```
[error] /Users/runner/work/directory-watcher/directory-watcher/core/src/main/java/io/methvin/watcher/visitor/DefaultFilesVisitor.java:21:1: cannot infer type arguments for java.nio.file.SimpleFileVisitor<T>
[error]   reason: '<>' with anonymous inner classes is not supported in -source 8
[error]     (use -source 9 or higher to enable '<>' with anonymous inner classes)
[error] SimpleFileVisitor<>
```

Hopefully we'll be able to drop jdk8 support in the next year or so, but at the moment directory-watcher is being used by other libraries that still need to support it.